ethereum/client.go

```diff
@@ -210,6 +210,87 @@ func toBlockNumArg(number *big.Int) string {
 	return hexutil.EncodeBig(number)
 }
 
+// Transaction returns the transaction response of the Transaction identified
+// by *RosettaTypes.TransactionIdentifier hash
+func (ec *Client) Transaction(
+	ctx context.Context,
+	blockIdentifier *RosettaTypes.BlockIdentifier,
+	transactionIdentifier *RosettaTypes.TransactionIdentifier,
+) (*RosettaTypes.Transaction, error) {
+	if transactionIdentifier.Hash != "" {
+		var raw json.RawMessage
+		err := ec.c.CallContext(ctx, &raw, "eth_getTransactionByHash", transactionIdentifier.Hash)
+		if err != nil {
+			return nil, fmt.Errorf("%w: transaction fetch failed", err)
+		} else if len(raw) == 0 {
+			return nil, ethereum.NotFound
+		}
+
+		// Decode transaction
+		var body rpcTransaction
+
+		if err := json.Unmarshal(raw, &body); err != nil {
+			return nil, err
+		}
+
+		var header *types.Header
+		if blockIdentifier.Hash != "" {
+			header, err = ec.blockHeaderByHash(ctx, blockIdentifier.Hash)
+		} else {
+			header, err = ec.blockHeaderByNumber(ctx, big.NewInt(blockIdentifier.Index))
```

Comment:
same here:
Use https://github.com/ethereum/go-ethereum/blob/0169d579d0eed4f6366697985a7b0f0b99402783/ethclient/ethclient.go#L180

Reply:
We are using native rpc calls rather then geth calls. We can do a complete overhaul separately